File: src/e-community/backend/cloud/e-community-cloud/e-community-cloud-lib/Endpoints/Dtos/BufferedMeterDataRTDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace e_community_cloud_lib.Endpoints.Dtos
{
    public class BufferedMeterDataRTDto
    {
        /// <summary>
        /// date and time of measuring point
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// how many smart meters are missing overall
        /// </summary>
        public int MissingSmartMeterCount { get; set; }

        /// <summary>
        /// how many smart meters are missing from a member
        /// </summary>
        public int MissingSmartMeterCountMember { get; set; }

        /// <summary>
        /// current power P+ of whole eCommunity
        /// </summary>
        public int ECommunityActivePowerPlus { get; set; }

        /// <summary>
        /// current power P- of whole eCommunity
        /// </summary>
        public int ECommunityActivePowerMinus { get; set; }

        /// <summary>
        /// current power R+ of whole eCommunity
        /// </summary>
        public int ECommunityReactivePowerPlus { get; set; }

        /// <summary>
        /// current power R- of whole eCommunity
        /// </summary>
        public int ECommunityReactivePowerMinus { get; set; }

        /// <summary>
        /// list of member's meter data
        /// </summary>
        public List<MeterDataRTDto> MeterDataMember { get; set; }
    }
}

File: src/e-community/android/compose/app/src/main/java/at/fhooe/ecommunity/data/remote/signalr/dto/BufferedMeterDataRTDto.kt
package at.fhooe.ecommunity.data.remote.signalr.dto

/**
 * real time meter data
 * @param timestamp date and time of measuring point
 * @param missingSmartMeterCount how many smart meters are missing overall
 * @param missingSmartMeterCountMember how many smart meters are missing from a member
 * @param eCommunityActivePowerPlus current power P+ of whole eCommunity
 * @param eCommunityActivePowerMinus current power P- of whole eCommunity
 * @param eCommunityReactivePowerPlus current power R+ of whole eCommunity
 * @param eCommunityReactivePowerMinus current power R- of whole eCommunity
 * @param meterDataMember meter data of member (per smart meter)
 */
data class BufferedMeterDataRTDto(val timestamp: String,
                                  val missingSmartMeterCount: Int,
                                  val missingSmartMeterCountMember: Int,
                                  val eCommunityActivePowerPlus: Int,
                                  val eCommunityActivePowerMinus: Int,
                                  val eCommunityReactivePowerPlus: Int,
                                  val eCommunityReactivePowerMinus: Int,
                                  val meterDataMember: List<MeterDataRTDto<Int>>?
)